content/en/blog/_posts/2022-04-12-prevent-unauthorised-volume-mode-conversion.md

```diff
@@ -0,0 +1,117 @@
+or `Patch` operations on `VolumeSnapshotContents`, which is a cluster-level resource.  
+It is upto the cluster administrator to provide these rights only to trusted users
+or applications, like backup vendors.
+
+If the alpha feature is [enabled](https://kubernetes-csi.github.io/docs/) in 
+`snapshot-controller` and `external-provisioner`, then unauthorised users will
+not be allowed to modify the volume mode of a PVC when it is being created from
+a `VolumeSnapshot`.
+
+To convert the volume mode, an authorised user must do the following:
+
+1. Identify the `VolumeSnapshot` that is to be used as the data source for a newly 
+created PVC in the given namespace. 
+2. Identify the `VolumeSnapshotContent` bound to the above `VolumeSnapshot`.
+
+```yaml
+$ kubectl get volumesnapshot -n <namespace>
+```
+
+3. Add the annotation [`snapshot.storage.kubernetes.io/allowVolumeModeChange`](/content/en/docs/reference/labels-annotations-taints/)
+to the `VolumeSnapshotContent`. 
+
+4. This annotation can be added either via software or manually by the authorised
+user. The `VolumeSnapshotContent` must look like below after this change:
+
+```yaml
+kind: VolumeSnapshotContent 
+metadata: 
+  annotations: 
+    - snapshot.storage.kubernetes.io/allowVolumeModeChange: "true"
+...
+```
+
+NOTE: For pre-provisioned `VolumeSnapshotContents`, the user has an additional 
+step of setting `spec.SourceVolumeMode` field to either `Filesystem` or `Block`,
+depending on the volume from which this snapshot was taken.
+
+An example is shown below:
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotContent
+metadata:
+  annotations:
+  - snapshot.storage.kubernetes.io/allowVolumeModeChange: "true"
+  name: new-snapshot-content-test
+spec:
+  deletionPolicy: Delete
+  driver: hostpath.csi.k8s.io
+  source:
+    snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  sourceVolumeMode: Filesystem
+  volumeSnapshotRef:
+    name: new-snapshot-test
+    namespace: default
+```
+
+Repeat (1)-(3) for all `VolumeSnapshotContents` whose volume mode needs to be 
+converted during a backup or restore operation.
+
+If the above annotation is present on a `VolumeSnapshotContent` object, 
+Kubernetes will not prevent the volume mode from being converted.
+Users should keep this in mind before they attempt to add the annotation 
+to any `VolumeSnapshotContent`. 
+
+
+### What's next
+
+[Enable this feature](https://kubernetes-csi.github.io/docs/) and let us know 
+what you think!
+
+We hope this feature causes no disruption to existing workflows while preventing
+malicious users from exploiting security vulnerabilities in their clusters. 
+
+For any issues, create a thread in the #sig-storage slack channel or an issue
+in the CSI external-snapshotter sidecar [repository](https://github.com/kubernetes-csi/external-snapshotter) 
+and assign it to [@RaunakShah](https://github.com/RaunakShah).
```

Comment:
Can you remove the "assign it to ..." part?

Reply:
done!